### ISSUE TITLE
ui: if a registry push fails, can't see that anywhere

### ISSUE THREAD

**Issue by mitchellh** [hashicorp/waypoint#1287]
This is a weird one kind of. If you have a registry configuration in your build, then the push operation isn't shown anywhere in the UI and so you can't see the failure. In the video below, we're seeing a situation where the _build_ succeeded (Docker image built) but the _push_ failed (auth error). Internally, Waypoint tracks these as two separate operations, so I can see why this is happening, just not sure what we should do from a UX perspective.

https://user-images.githubusercontent.com/1299/113939341-bca68500-97b0-11eb-9a75-65cc45438293.mp4

Waypoint.hcl:

```
app "learn-waypoint-lambda" {
  build {
    use "docker" {}

    registry {
      use "aws-ecr" {
        region = "us-west-2"
        repository = "learn-waypoint-lambda"
        tag = "howdy-waypoint-lambda"
      }
    }
  }
}
```

**Comment by mitchellh** [comment]
@gregone I think we can do this after 0.3.0, this feels a little more involved.

**Comment by gregone** [comment]
@mitchellh sounds good. In fact, after working on the log output component yesterday, I realized we're filtering the output quite a bit, and I need to do a dive into the log format to clean up some things, so this would be a good issue to do so.

**Comment by gregone** [comment]
https://github.com/hashicorp/waypoint/issues/1329

**Comment by jgwhite** [comment]
I’m going to take a quick look at this in light of #1668, and the new way we approach status indicators. I don’t think that change will have brought any accidental fixes, but may give us more of a steer on where to go with this. Either way, I propose deferring this until a later milestone (given that we’re approaching 0.5 and should be clearing 0.4.x items).

**Comment by jgwhite** [comment]
I’ve started to address this in #1840. It’s a significant change so I suggest targeting 0.5.0 or 0.5.x rather than 0.4.x, just to give us some more breathing room.

**Comment by jgwhite** [comment]
#1840 is looking good so I think a fix for this will land in 0.4.x (probably 0.4.2)

**Comment by jgwhite** [comment]
Closed by #1840
